### DISCUSSION TITLE
Is there a way to truncate the prompt other than printing “>”?

### DISCUSSION THREAD

**Discussion by jeffzyliu** [starship/starship#2445] [Ideas]
I think this isn't a feature (yet), so I'm putting this in the ideas section.

Sometimes I'm working on a narrower terminal window (or ssh'ing from my phone), and it's unfortunate that once I go into some dir with a few starship modules, the prompt line becomes too long, and it's just truncated to ">". In those situations, I'd much rather "prioritize" a select few modules to show when screen width is too narrow, rather than abandon the prompt entirely. (for example, I can't live without seeing the current wd in my prompt).

Is there a way to do this right now, and if not, is something like this in the works? Thanks :))

**Comment by davidkna** [comment]
Fish 3.2 fixes this by truncating the prompt instead.

**Reply by jeffzyliu** [reply]
Oh I see, thanks for clarifying! Sorry, thought this was a quirk with starship instead of fish. Should have guessed. Much better after updating.

**Reply by jeffzyliu** [reply]
Are there plans to implement starship-specific behavior to improve on this? I still think it might be cool to set "priority" levels to prompt modules so starship might do more sophisticated truncating itself if it runs out of screen space.

**Reply by davidkna** [reply]
There was some discussion in #691 on this and as it is currently implemented #2425 could also be a partial solution for this.
